src/core/server/legacy/legacy_service.ts

```diff
@@ -211,7 +216,13 @@ export class LegacyService implements CoreService {
           maxBytes: Number.MAX_SAFE_INTEGER,
         },
       },
-      handler: async ({ raw: { req, res } }, responseToolkit) => {
+      handler: async (request, responseToolkit) => {
+        const { req, res } = request.raw;
+        const authHeaders = setupDeps.core.http.auth.getAuthHeaders(request);
+        if (authHeaders) {
+          // some plugins in Legacy relay on headers.authorization presence
```

Comment:
for example https://github.com/restrry/kibana/blob/e751da3980f7de83a7248b080a3b876528b7e08c/x-pack/plugins/beats_management/server/lib/adapters/framework/hapi_framework_adapter.ts#L125